### PR TITLE
fix: Improve invalid provider version error

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Fixes https://github.com/cloudquery/cloudquery-issues/issues/387 (internal issue)

To reproduce:
1. `cloudquery init aws`
2. `git clone https://github.com/cloudquery/cq-provider-aws`
3. `make build` in the repo ⬆️ 
4. Copy complied provider to `.cq/providers/cloudquery/aws` and replace the existing one.
5. Run `cloudquery fetch --no-verify`

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: pkg/core/sync.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func Sync(ctx context.Context, sta *state.Client, pm *plugin.Manager, provider r
 	}
 
 	if want.ParsedVersion == nil {
-		return nil, diag.FromError(fmt.Errorf("failing provider with invalid version %q", provider.Version), diag.INTERNAL)
+		return nil, diag.FromError(fmt.Errorf("unsupported version %q for provider. If you're trying to debug a provider see https://docs.cloudquery.io/docs/developers/debugging", provider.Version), diag.USER)
```
Comment (roneli) [inline pkg/core/sync.go:71]: The message is 100% a better improvement, but I am not sure its correct. In line 65 we check that if parsed version is nil (so if it returns in 71 it should be nil) and provider version is unmanaged so it should not be nil in that case. 

Would not that returning version in GetProviderSchema was introduced later so might be that edge case maybe?

**Review by roneli — COMMENTED** [review]
LGTM, added a comment with some side thoughts about whether this solves the issue or not or just hides it under "user" type.

File: pkg/core/sync.go (hunk 1 above)
Comment (erezrokah) [reply pkg/core/sync.go:71]: >and provider version is unmanaged so it should not be nil in that case.

Agree this should not happen. I tested with unmanaged provider flow (with `CQ_REATTACH_PROVIDERS`) and couldn't reproduce.

The only way I could reproduce is by building the provider locally, overriding the official downloaded binary and running with `--no-verify` (without `CQ_REATTACH_PROVIDERS`).
Hence I'm assuming this is related somehow to a user trying to debug a provider by copying the binary under `.cq/providers`.

Comment (erezrokah) [reply pkg/core/sync.go:71]: >Would not that returning version in GetProviderSchema was introduced later so might be that edge case maybe?

Also I'm not sure I understand your intention with this comment

**Review by roneli — APPROVED** [review]

**Pull request merged**